File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/AllergyIntolerance.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class AllergyIntolerance : DomainResource, IResource
{
    public required Base.ResourceReference Patient { get; set; }
    public object? Onset    
    {
        get
        {
            if (OnsetRange is not null)
            {
                return OnsetRange;
            }
    
            if (OnsetAge is not null)
            {
                return OnsetAge;
            }
    
            if (OnsetPeriod is not null)
            {
                return OnsetPeriod;
            }
    
            if (OnsetString is not null)
            {
                return OnsetString;
            }
    
            if (OnsetDateTime is not null)
            {
                return OnsetDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.Range))
            {
                OnsetRange = (Base.Range)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OnsetAge = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                OnsetPeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OnsetString = (string)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OnsetDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public string[]? Category { get; set; }
    public string? Criticality { get; set; }
    public Base.CodeableConcept? ClinicalStatus { get; set; }
    public Base.Range? OnsetRange { get; set; }
    public string? OnsetAge { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Base.Period? OnsetPeriod { get; set; }
    public string? Type { get; set; }
    public Base.ResourceReference? Asserter { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? RecordedDate { get; set; }
    public string? OnsetString { get; set; }
    public Base.ResourceReference? Recorder { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? OnsetDateTime { get; set; }
    public string? LastOccurrence { get; set; }
    public Base.CodeableConcept? VerificationStatus { get; set; }
    public AllergyIntolerance_Reaction[]? Reaction { get; set; }

    public class AllergyIntolerance_Reaction : BackboneElement
    {
        public Base.Annotation[]? Note { get; set; }
        public string? Onset { get; set; }
        public string? Severity { get; set; }
        public Base.CodeableConcept? Substance { get; set; }
        public string? Description { get; set; }
        public Base.CodeableConcept? ExposureRoute { get; set; }
        public required Base.CodeableConcept[] Manifestation { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/AllergyIntolerance.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Age
from base import Period
from base import CodeableConcept
from base import Range
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class AllergyIntoleranceReaction(BackboneElement):
    note: Optional[List[Annotation]] = field(default_factory=list)
    onset: Optional[str] = None
    severity: Optional[str] = None
    substance: Optional[CodeableConcept] = None
    description: Optional[str] = None
    exposure_route: Optional[CodeableConcept] = None
    manifestation: list[CodeableConcept] = field(default_factory=list)

@dataclass(kw_only=True)
class AllergyIntolerance(DomainResource):
    patient: Reference
    category: Optional[List[str]] = field(default_factory=list)
    criticality: Optional[str] = None
    clinical_status: Optional[CodeableConcept] = None
    onset_range: Optional[Range] = None
    onset_age: Optional[Age] = None
    encounter: Optional[Reference] = None
    onset_period: Optional[Period] = None
    type: Optional[str] = None
    asserter: Optional[Reference] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    recorded_date: Optional[str] = None
    onset_string: Optional[str] = None
    recorder: Optional[Reference] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    onset_date_time: Optional[str] = None
    last_occurrence: Optional[str] = None
    verification_status: Optional[CodeableConcept] = None
    reaction: Optional[List[AllergyIntoleranceReaction]] = field(default_factory=list)